src/overview/overview.md

```diff
@@ -0,0 +1,643 @@
+        key:i3:e -> key2:i3:w [color="darkgreen", style="bold"]
+  }
+}
+```
+
+> **UNDER CONSTRUCTION** Actually explain the graph. Motivate why we are
+> describing areas backwards compared to the order of how passes are described
+> in conventional compilers, so the next section actually makes sense in terms
+> of its ordering. One benefit of doing ordering this way is that it kinda
+> forces you to have to talk about the inputs and outputs of each area, and also
+> kinda have to motivate _why_ each area exists -- what purpose do they serve?
+
+## Brief Summaries of the Areas
+
+> **UNDER CONSTRUCTION** This section lacks exhaustive list of areas and their
+> summaries. This should ideally be ordered based on a BFS traversal related to
+> the information dependency presented in the previous section.
+
+We now summarize areas following the order of their information dependency. To
+produce the final executable, we take one of the several code generation
+pathways. For example, to produce an executable following the LLVM pathway, we
+rely on the synthesis of LLVM IR from SSA code generation. Code generation in
+turn relies on information produced by Monomorphization Collection, and so on.
+We keep following the dependencies for the querified areas until we reach HIR
+Lowering, which depends on information from non-querified areas of the compiler.
+At that boundary, we make a distinction between the pull-based information
+dependency of querified areas versus push-based information dependency of the
+non-querified areas. We flip our direction, and trace from the source code until
+Late Name Resolution.
+
+To pass information between areas, the compiler uses various _Intermediate
+Representations_ (IRs), which provide common interfaces to allow areas to
+communicate with each other. These include but are not limited to:
+
+- Tokens
+- Abstract Syntax Tree (AST)
+- High-Level Intermediate Representation (HIR)
+- Mid-Level Intermediate Representation (MIR)
+
+These various IRs gradually transform surface Rust syntax by simplications of
+language constructs and enriches the information available to lower-level areas
+by many analyses.
+
+### Querified Areas
+
+> 🚧 **XXX DO NOT KEEP: EXAMPLE AREA SUMMARY** 🚧
+
+> - A succinct summary of the responsibility of the area.
+> - Introduce the inputs. Briefly motivate why the inputs are why they are.
+> - Motivate the area's existence -- what does it do to / with the inputs? Why?
+> - Introduce the outputs. Briefly motivate why the outputs are what they are.
+> - Want to know more? Link to details chapter here. Link to "Additional
+>   References" section below.
+
+> :warning: **Where does trait system come in?** / **Where do we describe the
+> trait system here?** AFAIK, it's both dependent and and provides information
+> to multiple areas. Link: <https://github.com/rust-lang/rust/pull/121848>
+
+#### Static-Single Assignment (SSA) Code Generation
```

Comment:
You're missing a section about codegen. Idk if it's querified or not

Reply:
~~We don't talk about codegen~~ Yeah I'm not completely sure about codegen atm.

Reply:
Stubbed out a section on the 3 codegen backends